validator/client/beacon-api/propose_beacon_block.go

```diff
@@ -62,7 +64,32 @@ func (c *beaconApiValidatorClient) proposeBeaconBlock(ctx context.Context, in *e
 	}
 
 	headers := map[string]string{"Eth-Consensus-Version": res.consensusVersion}
-	err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(res.marshalledJSON), nil)
+
+	// Try PostSSZ first with SSZ data
+	if res.marshalledSSZ != nil {
+		headers["Content-Type"] = "application/octet-stream"
+		_, _, err = c.jsonRestHandler.PostSSZ(ctx, endpoint, headers, bytes.NewBuffer(res.marshalledSSZ))
+		// If PostSSZ fails, fall back to JSON
+		if err != nil && res.marshalJSON != nil {
+			// Marshal JSON now that we need it
+			jsonData, jsonErr := res.marshalJSON()
+			if jsonErr != nil {
+				return nil, errors.Wrap(jsonErr, "failed to marshal JSON fallback")
+			}
+			// Reset headers for JSON
+			headers = map[string]string{"Eth-Consensus-Version": res.consensusVersion}
+			err = c.jsonRestHandler.Post(ctx, endpoint, headers, bytes.NewBuffer(jsonData), nil)
```

Comment:
should I have fallback at all? or is that dangerous?

Reply:
i'm going to make this more explicit on only falling back if they dont support ssz